Library/Homebrew/patch.rb

```diff
@@ -165,3 +167,18 @@ def inspect
     "#<#{self.class.name}: #{strip.inspect} #{url.inspect}>"
   end
 end
+
+
+# An abstract class representing a patch living alongside the formula.
+#
+# @api private
+class TapFilePatch < EmbeddedPatch
+  def initialize(strip, tap_file)
+    super(strip)
+    @str = File.open(tap_file).read
+  end
+
+  def contents
+    @str
+  end
```

Comment:
```suggestion
class TapFilePatch < EmbeddedPatch
  attr_reader :contents

  sig { params(strip: T.any(Symbol, String), tap_file: T.any(String, Pathname)).void }
  def initialize(strip, tap_file)
    super(strip)
    @contents = File.open(tap_file).read
  end
```

Note: I expect that the `File.open` may need to be a little more involved here to be able to find the patch in a suitable location. The ideal location in a tap would feel like a `Patches/` subdirectory in the root. Thoughts?

Reply:
I personally would prefer to keep the patches alongside the formula in the filesystem/editor... so would prefer not too have a patches folder. Instead Perhaps we can have:

```
Formula.rb
Formula.patch_name.diff
```
?

Reply:
If that's all you need then you can probably already do something like

```ruby
patch do
  url "file://#{__dir__}/Formula.patch_name.diff"
end
```
without having to make any changes to the formula DSL.

Reply:
yes

Reply:
Ah then never mind.